src/groupByKey.ts

```diff
@@ -2,6 +2,21 @@ type GroupsMap<T> = {
   [key: string]: T[];
 };
 
-export function groupByKey(items, key) {
-  // write code here;
+export function groupByKey<T>(items: T[], key: keyof T): GroupsMap<T> {
+  // const group = {} as GroupsMap<T>;
+
+  const group = items.reduce((books: GroupsMap<T>, book) => {
```

Comment:
```suggestion
  return items.reduce((books: GroupsMap<T>, book) => {
```
you can immediately return result of reduce